sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -202,13 +202,32 @@ std::string GetHeadersAsString(Azure::Core::Http::Request const& request)
 {
   std::string requestHeaderString;
 
-  for (auto const& header : request.GetHeaders())
+  // request.GetHeaders() aggregates the pre- and post-retry headers into a single map. Capture it
+  // so we don't recalculate the merge multiple times.
+  auto requestHeaders = request.GetHeaders();
+
+  for (auto const& header : requestHeaders)
   {
     requestHeaderString += header.first; // string (key)
     requestHeaderString += ": ";
     requestHeaderString += header.second; // string's value
     requestHeaderString += "\r\n";
   }
+
+  // The test recording infrastructure requires that a Patch verb have a Content-Length header,
+  // because it does not distinguish between requests with and without a body if there's no
+  // Content-Length header.
+  if (request.GetMethod() == HttpMethod::Patch)
```

Comment:
Do we want to align with libcurl implementation? 

https://github.com/Azure/azure-sdk-for-cpp/blob/50c18516df110058e1faac27b38360bc56db7c6e/sdk/core/azure-core/src/http/curl/curl.cpp#L428

Libcurl adds `Content-Length: 0` if the request method is not `Get`, `Head`, `Delete`?

Reply:
I also think we should align.

Reply:
I completely disagree.

Libcurl is a 100% hack which implements the entirety of the HTTP client in our code.
WinHTTP is a production ready HTTP client. We *NEVER* want to second guess the winhttp code.

If there is a disagreement between libcurl and winhttp, WInHTTP wins.